gammapy/scripts/analysis.py

```diff
@@ -0,0 +1,76 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Session class driving the high-level interface API"""
+import jsonschema
+import logging
+from ..utils.scripts import read_yaml
+from astropy import units as u
+from pathlib import Path
+
+__all__ = ["Analysis"]
+
+log = logging.getLogger(__name__)
+CONFIG_PATH = Path(__file__).resolve().parent / "config"
+CONFIG_FILE = CONFIG_PATH / "default.yaml"
+SCHEMA_FILE = CONFIG_PATH / "schema.yaml"
+
+
+class Analysis():
+    """High-level interface analysis session class.
+
+    The Analysis class drives an analysis working session using the high-level interface
+    API. It is initialized by default with a set of configuration parameters and values
+    declared in an internal configuration YAML file.
+
+    Parameters
+    ----------
+    configfile : string
+        The name of a user defined configuration file.
+    """
+
+    def __init__(self, configfile=CONFIG_FILE, **kwargs):
+        self.config = dict()
+        if configfile:
+            self.configfile = configfile
+            self.config = read_yaml(self.configfile)
+        if len(kwargs):
+            self.config.update(**kwargs)
+        self._set_logging()
+
+    def validate_config(self):
+        """Validate config parameters against schema."""
+        schema = read_yaml(SCHEMA_FILE)
+        try:
+            jsonschema.validate(self.config, schema, _gp_validator)
+        except jsonschema.exceptions.ValidationError as ex:
```

Comment:
Please remove try-except, or add a test for the except case. From just reading it, I have no idea if `ex.schema_path[-2]` will work or not, this kind of code that was only tested manually once is often where bugs hide.